File: packages/server/internal/model/material.go
package model

import "time"

// DeviceMaterial 设备维修耗材台账
// 记录路灯设备使用/备用的耗材（灯泡型号、电源、控制器等），用于维修派单参考
type DeviceMaterial struct {
	ID            uint      `json:"id" gorm:"primaryKey"`
	DeviceHwID    uint32    `json:"device_hw_id" gorm:"index;comment:设备硬件ID"`
	Name          string    `json:"name" gorm:"size:64;comment:耗材名称"`
	PartNo        string    `json:"part_no" gorm:"size:64;comment:型号/料号"`
	Spec          string    `json:"spec" gorm:"size:128;comment:规格参数"`
	Quantity      int       `json:"quantity" gorm:"default:0;comment:当前库存"`
	Unit          string    `json:"unit" gorm:"size:16;comment:单位(个/支/套)"`
	Threshold     int       `json:"threshold" gorm:"default:0;comment:库存预警阈值"`
	LastChangedAt *time.Time `json:"last_changed_at" gorm:"comment:最近变动时间"`
	Note          string    `json:"note" gorm:"type:text;comment:备注"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

// TableName 指定表名
func (DeviceMaterial) TableName() string {
	return "device_materials"
}

File: packages/server-java/src/main/java/com/tsloms/server/model/Material.java
// 物料档案实体：逐列对齐 Go 版 Material（表 materials）。
package com.tsloms.server.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import java.time.Instant;

/** 物料台账：同一物料可多处使用。 */
@Entity
@Table(name = "materials",
        uniqueConstraints = @UniqueConstraint(name = "uk_materials_code", columnNames = "code"),
        indexes = {
                @Index(name = "idx_materials_name", columnList = "name"),
                @Index(name = "idx_materials_category", columnList = "category"),
                @Index(name = "idx_materials_device_hw", columnList = "device_hw_id"),
                @Index(name = "idx_materials_supplier", columnList = "supplier_id"),
        })
public class Material {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    @Column(name = "code", nullable = false, length = 32)
    public String code;

    @Column(name = "name", nullable = false, length = 64)
    public String name;

    /** 分类：灯组/电源/线材等。 */
    @Column(name = "category", length = 32)
    public String category;

    @Column(name = "spec", length = 128)
    public String spec;

    @Column(name = "unit", length = 16)
    public String unit;

    @Column(name = "unit_price", nullable = false)
    public Double unitPrice = 0.0;

    @Column(name = "stock", nullable = false)
    public Integer stock = 0;

    /** 低库存预警阈值。 */
    @Column(name = "threshold", nullable = false)
    public Integer threshold = 0;

    /** 关联设备 ID（可空）。 */
    @Column(name = "device_hw_id", length = 64)
    public String deviceHwId;

    @Column(name = "supplier_id")
    public Long supplierId;

    @Column(name = "note", columnDefinition = "text")
    public String note;

    @Column(name = "status", nullable = false, length = 16)
    public String status = "active";

    @Column(name = "created_at", nullable = false)
    public Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    public Instant updatedAt;

    @PrePersist
    void onCreate() {
        Instant now = Instant.now();
        if (createdAt == null) {
            createdAt = now;
        }
        if (updatedAt == null) {
            updatedAt = now;
        }
    }

    @PreUpdate
    void onUpdate() {
        updatedAt = Instant.now();
    }
}
